framework/ui/src/main/java/com/google/android/catalog/framework/ui/components/AdaptivePane.kt

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.android.catalog.framework.ui.components
+
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.PaddingValues
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.unit.dp
+
+@Composable
+internal fun AdaptivePane(
+    paddingValues: PaddingValues,
+    start: @Composable () -> Unit,
+    end: @Composable () -> Unit
+) {
+    if (isExpandedScreen()) {
+        Row(Modifier.fillMaxSize()) {
+            Box(
+                modifier = Modifier
+                    .fillMaxSize()
+                    .weight(1.3f)
+            ) {
+                start()
+            }
+            Box(
+                modifier = Modifier
+                    .fillMaxSize()
+                    .padding(paddingValues)
+                    .padding(top = 4.dp, end = 4.dp, bottom = 4.dp)
+                    .clip(RoundedCornerShape(16.dp))
+                    .background(MaterialTheme.colorScheme.surfaceVariant)
+                    .weight(2f)
+            ) {
+                end()
+            }
+        }
+    } else {
+        start()
```

Comment:
It looks like the behavior here is that if a sample is selected and both the list and detail are open, and then if the window becomes smaller, we'll revert to showing just a list.

I think that's fine for this sample case, given that some of the samples can't be open in the detail directly, but normally for a list-detail view it'd be good to keep the selected detail open ([ref](https://m3.material.io/foundations/layout/canonical-layouts/list-detail#19de8904-8f61-4af7-8b5a-78c9edeb62b8))

Reply:
Good point but I think for this use case it makes sense to keep the behavior like this